src/coreclr/jit/assertionprop.cpp

```diff
@@ -5852,18 +5902,24 @@ void Compiler::optVnNonNullPropCurStmt(BasicBlock* block, Statement* stmt, GenTr
 //
 // Description:
 //    An unified value numbering based assertion prop visitor that
-//    performs non-null and constant assertion propagation based on
+//    performs non-null, constant assertion and other types of propagation based on
 //    value numbers.
 //
 /* static */
 Compiler::fgWalkResult Compiler::optVNAssertionPropCurStmtVisitor(GenTree** ppTree, fgWalkData* data)
 {
-    VNAssertionPropVisitorInfo* pData = (VNAssertionPropVisitorInfo*)data->pCallbackData;
-    Compiler*                   pThis = pData->pThis;
+    auto*     pData = (VNAssertionPropVisitorInfo*)data->pCallbackData;
+    Compiler* pThis = pData->pThis;
 
     pThis->optVnNonNullPropCurStmt(pData->block, pData->stmt, *ppTree);
 
-    return pThis->optVNConstantPropCurStmt(pData->block, pData->stmt, *ppTree);
+    bool         changed;
+    fgWalkResult walkResult = pThis->optVNConstantPropCurStmt(pData->block, pData->stmt, *ppTree, &changed);
+    if (!changed && (walkResult == WALK_CONTINUE))
```

Comment:
Is there ever a case when `walkResult` other than `WALK_CONTINUE` will be returned when `changed` is `false`?

Reply:
not now, tried to make it more agile, probably can convert to an assert for now untill it's changed (for TP)